lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -257,8 +256,17 @@ def extract_student(student, features):
                             for feature in student_features)
         profile = student.profile
         if profile is not None:
-            profile_dict = dict((feature, extract_attr(profile, feature))
-                                for feature in profile_features)
+            profile_dict = dict()
+
+            for feature in profile_features:
+                if feature in ['user_age', 'region', 'profession', 'gender']:
+                    get_display_name = getattr(profile, "get_{field_name}_display".format(field_name=feature))
```

Comment:
get_display_name - would avoid creating a new variable if it is not used more than once:
```
profile_attr = getattr(profile, "get_{field_name}_display".format(field_name=feature))()
``` 
just the comment it is up to you

Reply:
I changed it